test/integration/support/test_server.ex

```diff
@@ -0,0 +1,34 @@
+defmodule ChromicPDF.TestServer do
+  @moduledoc false
+
+  use Plug.Router
+
+  @port 44_285
+
+  @cowboy_opts [
+    http: [port: @port],
+    https: [
+      port: @port,
```

Comment:
feels a bit weird how `http` and `https` apparently run on the same port. How can this even work?

Reply:
They never run at the same time :)

can make it different ports though, if that makes it more clear

Reply:
Can the ports be somehow auto assigned to a free one, like sockets to when assigning `0` as a port? Would be great.

Reply:
🤔  good idea. It is possible, but I don't know how to get ahold of the used port unfortunately. plug_cowboy doesn't have any functions for it. So, yes, nice to have, but IMO not necessary to put a lot of time into.